### PR TITLE
GEODE-4384: gfsh command to destroy jndi binding

### PR DESCRIPTION
**Pull request by sboorlagadda**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      @CliOption(key = JNDI_NAME, mandatory = true, help = JNDI_NAME__HELP) String jndiName)
+      throws IOException, SAXException, ParserConfigurationException, TransformerException {
+
+    Result result;
+    boolean persisted = false;
+    ClusterConfigurationService service = getSharedConfiguration();
+    if (service != null) {
+      Element existingBinding =
+          service.getXmlElement("cluster", "jndi-binding", "jndi-name", jndiName);
+      if (existingBinding == null)
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Braces

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        return ResultBuilder.createUserErrorResult(
+            CliStrings.format("Jndi binding with jndi-name \"{0}\" does not exists.", jndiName));
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Spelling `exists` -> `exist`

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      persisted = true;
+    }
+
+    Set<DistributedMember> targetMembers = findMembers(null, null);
+    if (targetMembers.size() > 0) {
+      List<CliFunctionResult> jndiCreationResult =
+          executeAndGetFunctionResult(new DestroyJndiBindingFunction(), jndiName, targetMembers);
+      return buildResult(jndiCreationResult);
+    } else {
+      if (persisted)
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Braces again

File: geode-core/src/main/java/org/apache/geode/internal/jndi/JNDIInvoker.java [hunk 4]
```diff
@@ -360,6 +361,18 @@ public static void mapDatasource(Map map, List props) {
     }
   }
 
+  public static void unMapDatasource(String jndiName) throws NamingException {
+    ctx.unbind("java:/" + jndiName);
+    for (Iterator it = dataSourceList.iterator(); it.hasNext();) {
+      Object obj = it.next();
+      if (obj instanceof AbstractDataSource)
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/jndi/JNDIInvoker.java]: Just a minor nit - please add block braces

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java [hunk 5]
```diff
@@ -162,7 +162,9 @@ public Result createJDNIBinding(
     ClusterConfigurationService service = getSharedConfiguration();
 
     if (service != null) {
-      if (isBindingAlreadyExists(jndiName))
+      Element existingBinding =
+          service.getXmlElement("cluster", "jndi-binding", "jndi-name", jndiName);
+      if (existingBinding != null)
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java]: Braces again

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.geode.management.internal.configuration.utils.XmlUtils;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+public class DestroyJndiBindingCommand implements GfshCommand {
+  static final String CREATE_JNDIBINDING = "destroy jndi-binding";
+  static final String CREATE_JNDIBINDING__HELP =
+      "Destroy a jndi binding that holds the configuration for the XA datasource.";
+  static final String JNDI_NAME = "name";
+  static final String JNDI_NAME__HELP = "Name of the binding to be created.";
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Probably should be `destroyed` and not `created`?

Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Done

File: geode-core/src/main/java/org/apache/geode/internal/jndi/JNDIInvoker.java (hunk 4 above)
Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/internal/jndi/JNDIInvoker.java]: Done

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java (hunk 3 above)
Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Done

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java (hunk 2 above)
Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Done

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java (hunk 1 above)
Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyJndiBindingCommand.java]: Done

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java (hunk 5 above)
Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java]: Done

**Pull request merged**